packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/index.js

```diff
@@ -1,60 +1,51 @@
-import React, { useReducer, useState } from "react";
+import React from "react";
 import PropTypes from "prop-types";
 import { OnChange } from "react-final-form-listeners";
 import { Section, Container, Wrapper } from "@cdt/ui";
 
-import { StepReducer, getInitialSteps } from "./reducer";
+import { stepReducer, initialSteps } from "./stepReducer";
 import { Wizard } from "./Wizard";
 
 function CalculateurIndemnite() {
-  const initialSteps = getInitialSteps();
-  const [steps, dispatch] = useReducer(StepReducer, initialSteps);
-  const [values, setValues] = useState({});
   /**
-   * The rules defined here allows to manage additionnal steps to the form
+   * The rules defined here allows to manage additionnal steps to the wizard
    */
-  const rules = [
-    <OnChange key="rule-same-salaire" name="hasSameSalaire">
-      {value =>
-        value === false
-          ? dispatch({ type: "add_primes" })
-          : dispatch({ type: "remove_primes" })
-      }
-    </OnChange>,
-    <OnChange key="rule-branche" name="branche">
-      {async value => {
-        if (value) {
-          const module = await import(`./ccn/${value}`);
-          const steps = module.steps.filter(({ condition = () => true }) =>
-            condition(values)
-          );
-          dispatch({ type: "add_branche", payload: steps });
-        } else {
-          dispatch({ type: "remove_branche" });
+  const Rules = ({ values, dispatch }) => (
+    <>
+      <OnChange key="rule-same-salaire" name="hasSameSalaire">
+        {value =>
+          value === false
+            ? dispatch({ type: "add_primes" })
+            : dispatch({ type: "remove_primes" })
         }
-      }}
-    </OnChange>
-  ];
-
-  // when at the end, the form is submitted
-  // we reset the form data
-  const onSubmit = () => {
-    dispatch({
-      type: "reset",
-      payload: getInitialSteps()
-    });
-  };
+      </OnChange>
+      <OnChange key="rule-branche" name="branche">
+        {async value => {
+          if (value) {
+            const module = await import(`./ccn/${value}`);
+            const steps = module.steps.filter(({ condition = () => true }) =>
+              condition(values)
+            );
+            dispatch({ type: "add_branche", payload: steps });
+          } else {
+            dispatch({ type: "remove_branche" });
+          }
+        }}
+      </OnChange>
+    </>
+  );
 
   return (
     <Section>
       <Container>
         <Wrapper size="large" variant="light">
           <h1>Calculateur d&apos;indemnités de licenciement</h1>
           <Wizard
-            steps={steps}
-            onSubmit={onSubmit}
-            onUpdate={setValues}
-            rules={rules}
+            stepReducer={stepReducer}
```

Comment:
So the main idea of the fix is to delegate the handling of the steps to the wizard. So now, the steps are managed inside of it and not above it, and therefore, fix the rerendering issue when changing them.

Reply:
But we can still configure it as before by providing initialStepIndex, InitialValues, initialSteps and rules.

Reply:
I wonder if we can not group Rules and stepReducer since they are really inter-dependant

Reply:
I like the idea ! But I need to figure out how

Reply:
no problem, we can do it later :) we can add a reminder

Reply:
I'm going to create an issue